byway/src/main/java/com/google/sps/servlets/PlacesServlet.java

```diff
@@ -52,4 +64,35 @@ private String formatLocation(PlaceType location) {
     place = place.replace('_', ' ');
     return place;
   }
+
+  /**
+   * Temporary setup to find entity with hard-coded id value. Set their interests.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query specificTrip =
+      new Query("trip")
+        .setFilter(new FilterPredicate("id", FilterOperator.EQUAL, 2452));
+    PreparedQuery pq = datastore.prepare(specificTrip);
+
+    // Guaranteed to have one entity, since id would be unique.
+    Entity trip = pq.asSingleEntity();
+    String interestsAsString = request.getParameter("interests");
+    addInterestsForTrip(interestsAsString, trip);
+  }
+
+
+  /**
+   * Creates a list of interests from the JSON string passed in and adds it
+   * to this user's Entity object as a separate property.
+   * @param interestsAsString JSON String containing interests.
+   * @param currentUser       Entity with user information.
+   * @param datastore         DatastoreService with all users.
+   */
+  private void addInterestsForTrip(String interestsAsString, Entity trip) throws IOException {
+
+    ArrayList<String> interests = gson.fromJson(interestsAsString, ArrayList.class);
+    trip.setProperty("interests", interests);
+    datastore.put(trip);
+  }
```

Comment:
I'd try to break this down into two operations a little differently. Right now you have the caller get the parameter from the request, then this function combines the operations  of (parse json string, apply update to entity). To me, this doesn't seem like a very logical grouping, since it's very specific to this use case:

caller: (Get json string, Get entity)
function: (Parse json string, update entity)

Notice how this grouping has both the caller and the called function operating with both json strings and entities. Additionally, the caller doesn't have any flexibility regarding how they use the parsed data.

Consider this alternate grouping of functionality:

function 1: (Get json string, Parse json string)
function 2: (Get entity, update entity)

Each function operates on only a single type of data, either json or entities, but *not* both, with the intermediary between them being `ArrayList<String>`. this separation of which kind of data the functions operate on makes it much easier to generalize them in the future, if you find yourself doing similar work in many places.

Reply:
I agree this setup is somewhat odd/very specific. Fixed it to separate the tasks more reasonably